### ISSUE TITLE
Add NNX support for legacy `jax.random.PRNGKey()`

### ISSUE THREAD

**Issue by cisprague** [google/flax#4231]
Currently, it doesn't seem possible to straightforwardly checkpoint (with Orbax) an NNX module that includes random keys (like with dropout), see https://github.com/google/orbax/issues/1105#issue-2488663946. This seems to be due to the new JAX random key type (https://jax.readthedocs.io/en/latest/jep/9263-typed-keys.html), which is used here (https://github.com/google/flax/blob/fc19c5dcc136b17d7a2c406831126d93bbdac6d7/flax/nnx/rnglib.py#L186) in NNX. Although Orbax has added individual support for the new type (see https://github.com/google/orbax/issues/620), saving `nnx.state(model)` that includes `dtype=key<fry>` doesn't seem to be possible.

**Comment by cgarciae** [comment]
Hey @cisprague, maybe you can convert to the old format before serializing?
You could use something like:

```python
def get_key_data(x):
  # use jax.random.key_data
  
state = jax.tree.map(get_key_data, state)
```
See [PRNGKeys](https://jax.readthedocs.io/en/latest/jax.random.html#prng-keys)

**Comment by shanksauce** [comment]
Following Cristian's recommendation, FWIW, this worked for me until the core issue is resolved.

```
# saving
_, state = nnx.split(model)

state = jax.tree.map(
    lambda x: jax.random.key_data(x) if x.dtype.name == "key<fry>" else x,  # ☹️
    state
)

checkpoint_manager.save(0, args=ocp.args.StandardSave(state))

# restoring
abstract_model = nnx.eval_shape(lambda: model)

graphdef, abstract_state = nnx.split(abstract_model)

abstract_state = jax.tree.map(
    lambda x: None if x.dtype.name == "key<fry>" else x, 
    abstract_state
)

res = checkpoint_manager.restore(0, args=ocp.args.StandardRestore(abstract_state))

model = nnx.merge(graphdef, res["state"])
```
